operator/controllers/provision.go

```diff
@@ -0,0 +1,246 @@
+
+	return nil
+}
+
+func (r *FlowTestReconciler) deploySlicedFlows(ctx context.Context, extraLabels map[string]string) error {
+	logger := log.FromContext(ctx)
+	flowTest := ctx.Value("flowTest").(loggingplumberv1alpha1.FlowTest)
+
+	var referenceFlow flowv1beta1.Flow
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: flowTest.Spec.ReferenceFlow.Namespace,
+		Name:      flowTest.Spec.ReferenceFlow.Name,
+	}, &referenceFlow); err != nil {
+		return err
+	}
+
+	i := 0
+
+	flowTemplate, outTemplate := banzaiTemplates(referenceFlow, flowTest, extraLabels)
+
+	for x := 1; x <= len(referenceFlow.Spec.Match); x++ {
+		targetFlow := *flowTemplate.DeepCopy()
+		targetOutput := *outTemplate.DeepCopy()
+
+		targetFlow.ObjectMeta.Name = fmt.Sprintf("%s-%d-match", flowTest.ObjectMeta.UID, i)
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "match"
+
+		targetOutput.ObjectMeta.Name = fmt.Sprintf("%s-%d-match", flowTest.ObjectMeta.UID, i)
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "match"
+		targetOutput.Spec.HTTPOutput.Endpoint = fmt.Sprintf("%s/%s/", targetOutput.Spec.HTTPOutput.Endpoint, targetFlow.ObjectMeta.Name)
+
+		targetFlow.Spec.LocalOutputRefs = []string{targetOutput.ObjectMeta.Name}
+
+		targetFlow.Spec.Match = append(targetFlow.Spec.Match, referenceFlow.Spec.Match[:x]...)
+
+		if err := r.Create(ctx, &targetFlow); err != nil {
+			logger.Error(err, fmt.Sprintf("failed to deploy Flow #%d for %s", i, referenceFlow.ObjectMeta.Name))
+			return err
+		}
+		if err := r.Create(ctx, &targetOutput); err != nil {
+			logger.Error(err, fmt.Sprintf("failed to deploy Flow #%d for %s", i, referenceFlow.ObjectMeta.Name))
+			return err
+		}
+		logger.V(1).Info("deployed match slice", "test-id", i)
+		i++
+	}
+
+	for x := 1; x <= len(referenceFlow.Spec.Filters); x++ {
+		targetFlow := *flowTemplate.DeepCopy()
+		targetOutput := *outTemplate.DeepCopy()
+
+		targetFlow.ObjectMeta.Name = fmt.Sprintf("%s-%d-filture", flowTest.ObjectMeta.UID, i)
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "filter"
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+
+		targetOutput.ObjectMeta.Name = fmt.Sprintf("%s-%d-filture", flowTest.ObjectMeta.UID, i)
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "filter"
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+		targetOutput.Spec.HTTPOutput.Endpoint = fmt.Sprintf("%s/%s/", targetOutput.Spec.HTTPOutput.Endpoint, targetFlow.ObjectMeta.Name)
+
+		targetFlow.Spec.LocalOutputRefs = []string{targetOutput.ObjectMeta.Name}
+
+		targetFlow.Spec.Match = nil
+
+		targetFlow.Spec.Filters = referenceFlow.Spec.Filters[:x]
+
+		if err := r.Create(ctx, &targetFlow); err != nil {
```

Comment:
we should create the output before the flow.  This caused issues in older versions of banzai logging